haskell.nix/README.md

```diff
@@ -56,3 +56,105 @@ Haskell application and library templates for Buildkite, Gitlab or GitHub CI usi
     2. If your project contains bash scripts, uncomment related lines in `flake.nix` and in the pipeline configuration. If you don't need `shellcheck`, remove those lines.
 
 - Enjoy working CI!
+
+## Impure tests
+
+By default, these templates run Haskell tests in a pure nix builds, so it is not possible to access external resources such as a database or docker socket.
+
+To get around this, you first need to provide a nix shell with the test executable along with other packages needed for the test environment.
+
+Here, as an example, we provide a nix shell with the `pataq-package-test` executable, as well as two additional packages: `ephemeralpg`, used to create the ephemeral database, and `postgresql`, used to create the database schema:
+
+```nix
+ devShells = {
+   impure-tests = pkgs.mkShell {
+     buildInputs = [
+       hs-pkg.components.tests.pataq-package-test
+         pkgs.ephemeralpg # we need pg_tmp to create an ephemeral db
+         pkgs.postgresql # we need psql to create the db schema
+       ];
+     };
+   };
+```
+
+We then create a bash script that sets up the required environment and runs the `pataq-package-test`:
+
+```bash
+#!/usr/bin/env bash
+
+# Set up the necessary environment
+set -euo pipefail
+export LANG="C.UTF-8"
+export LC_ALL="C.UTF-8"
+PSQL_CONNECTION_STRING=$(pg_tmp -t)
+echo "Ephemeral PostgreSQL connection string: $PSQL_CONNECTION_STRING"
+
+psql "$PSQL_CONNECTION_STRING" -f ./pataq-schema.sql
+
+# some explanation for the env var magic below https://stackoverflow.com/questions/918886/how-do-i-split-a-string-on-a-delimiter-in-bash
+portAndDb=${PSQL_CONNECTION_STRING##*:}
+export POSTGRES_PORT_TEST=${portAndDb%%/*}
```

Comment:
AFAIU, these env variables are specific to a given test-suite, perhaps, I think it's worth mentioning that they're supposed to be used by the test suite, which is one way to provide DB connection details to tests.
Additionally, VM-based tests don't use any env variables, which might be a bit confusing